### PR TITLE
Project refactor and update dependencies

### PR DESCRIPTION
**Pull request by uF4No**
**This is a WIP PR until we merge the correspondent changes in the docs** ([PR](https://github.com/matter-labs/zksync-web-v2-docs/pull/113))

Re organization of the code to match new proposed version of the tutorial 
Updates dependencies

### CODE REVIEW

File: L2-counter/deploy/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { utils, Wallet } from 'zksync-web3';
+import * as ethers from 'ethers';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+
+const GOVERNANCE_ADDRESS = '0x3C4Ac9AfadB3F345744248Ec97DC3051c63d7c00';
+
+// An example of a deploy script that will deploy and call a simple contract.
```
Comment (barakshani) [inline L2-counter/deploy/deploy.ts:8]: "will be deployed" or "we/you will deploy"

File: L2-counter/hardhat.config.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require('@matterlabs/hardhat-zksync-deploy');
+require('@matterlabs/hardhat-zksync-solc');
+
+module.exports = {
+  zksolc: {
+    version: '1.2.0',
+    compilerSource: 'binary', // use "docker" if you dont have zksolc installed globally
```
Comment (barakshani) [inline L2-counter/hardhat.config.ts:7]: u -> U

File: L2-counter/scripts/increment-counter.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  );
+
+  // Encoding the tx data the same way it is done on Ethereum.
+  const counterInterface = new ethers.utils.Interface(COUNTER_ABI);
+  const data = counterInterface.encodeFunctionData('increment', []);
+
+  // The price of the L1 transaction requests depends on the gas price used in the call
+  const gasPrice = await l1Provider.getGasPrice();
+
+  // Here we define the constant for ergs limit.
```
Comment (barakshani) [inline L2-counter/scripts/increment-counter.ts:35]: why all of a sudden periods at the end of the comment?

File: L2-counter/deploy/deploy.ts (hunk 1 above)
Comment (uF4No) [reply L2-counter/deploy/deploy.ts]: fixed in c524d3a1470771290f717b27c219f20ecab10e72

File: L2-counter/hardhat.config.ts (hunk 2 above)
Comment (uF4No) [reply L2-counter/hardhat.config.ts]: fixed in c524d3a1470771290f717b27c219f20ecab10e72

File: L2-counter/scripts/increment-counter.ts (hunk 3 above)
Comment (uF4No) [reply L2-counter/scripts/increment-counter.ts]: fixed in c524d3a1470771290f717b27c219f20ecab10e72

File: L2-counter/deploy/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+
+const GOVERNANCE_ADDRESS = '<GOVERNANCE-CONTRACT-ADDRESS>';
+
+// An example script to deploy and call a simple contract
+export default async function (hre: HardhatRuntimeEnvironment) {
+  console.log(`Running deploy script for the Counter contract`);
+
+  // Initialize the wallet.
```
[the remark this answers is not in the document]
Comment (uF4No) [reply L2-counter/deploy/deploy.ts]: all updated in a96b06d8b5c1d5b2d6e907d3876c38bc6b175916

**Review by barakshani — APPROVED** [review]

**Pull request merged**